bindings/pyroot/pythonizations/python/ROOT/_pythonization/_tf1.py

```diff
@@ -0,0 +1,140 @@
+# Author: Aaron Jomy CERN 08/2024
+
+################################################################################
+# Copyright (C) 1995-2022, Rene Brun and Fons Rademakers.                      #
+# All rights reserved.                                                         #
+#                                                                              #
+# For the licensing terms see $ROOTSYS/LICENSE.                                #
+# For the list of contributors see $ROOTSYS/README/CREDITS.                    #
+################################################################################
+
+r"""
+/**
+\class TF1
+\brief \parblock \endparblock
+\htmlonly
+<div class="pyrootbox">
+\endhtmlonly
+## PyROOT
+
+The TF1 class has several additions for its use from Python, which are also
+available in its subclasses TF2 and TF3.
+
+First, TF1 instance can be inititialised with user-defined Python functions. Given a functor object or a pycallable,
+the following can performed:
+
+\code{.py}
+def func(x, pars):
+    return pars[0] * x[0] * x[0] + x[1] * pars[0]
+
+my_func = ROOT.TF1("my_func", func, -10, 10, npar=2, ndim=2)
+\endcode
+
+Second, after performing the initialisation with a Python functor, the TF1 instance can be evaluated using the Pythonized
+`TF1::EvalPar` function. The pythonization allows passing in the expected data(x {1, 2, .. ndims}) or a dataset (x { {1, 2, .. ndims}, {1, 2, .. ndims}...})
+as 1D or 2D NumPy arrays.
+
+The following example shows how we can create a TF1 instance with a Python function and evaluate it on both a single set of x variables or a dataset:
+
+\code{.py}
+import ROOT
+
+def func(x, pars):
+    return pars[0] * x[0] * x[0] + x[1] * pars[0]
+
+my_func = ROOT.TF1("my_func", func, -10, 10, npar=2, ndim=2)
+
+x1 = np.array([2., 1]) # instance of x variables
+x = np.array([[2., 1], # dataset of x variables
+              [1., 3],
+              [2., 1],
+              [4., 2]])
+
+pars = np.array([2.])
+
+y1 = myfunc.EvalPar(x1, pars)
+
+y = myfunc.EvalPar(x, pars)
+
+assert y1 == y[0] 
+\endcode
+
+This functionality also supports passing sliced dataset or input variables where the array data is non-contiguous. An example:
+
+\code{.py}
+import ROOT
+import numpy as np
+
+rtf1_coulomb = ROOT.TF1("my_func", pyf_tf1_coulomb, -10, 10, ndims = 2, npars = 3)
+
+# x dataset: 5 pairs of particle charges
+x = np.array([
+    [1.0, 10, 2.0],
+    [1.5, 10, 2.5],
+    [2.0, 10, 3.0],
+    [2.5, 10, 3.5],
+    [3.0, 10, 4.0]
+])
+
+params = np.array([
+    [1.0],       # Distance between charges r
+    [8.99e9],    # Coulomb constant k (in N·m²/C²)
+    [0.1]        # Additional factor for modulation
+])
+
+# Slice to avoid the dummy column of 10's
+res = rtf1_coulomb.EvalPar(x[:, ::2], params)
+        
+\endcode
+
+<em>Please note</em> that the constructor must be initialised with `ndim` equal to or more than the size of each x variable instance.
```

Comment:
I believe it would be worth investigating how difficult it would be to provide a default `ndim == len(x)` internally so the user is not forced to specify.

Reply:
I tried to achieve this but we don't have a `SetNdim` function. Based on my understanding, then it can only depend on the size passed to the constructor since the data is only passed at evaluation time.  

I have implemented the same and it seems to work. I was first weary of exposing this as public, but as per my understanding, calling `Update` should reset the histogram. The constructor that uses a formula definition does change `fNdim` internally based on the `TFormula` used.